runner/ollamarunner/cache.go

```diff
@@ -260,11 +268,25 @@ func (c *InputCache) ShiftCacheSlot(slot *InputCacheSlot, numKeep int32) error {
 	slog.Debug("context limit hit - shifting", "id", slot.Id, "limit", c.numCtx, "input", len(slot.Inputs),
 		"keep", numKeep, "discard", discard)
 
-	// TODO (jessegross): KV cache removal can fail for certain types of models
 	if c.cache != nil {
 		err := c.cache.Remove(slot.Id, numKeep, numKeep+discard)
 		if err != nil {
-			return fmt.Errorf("unable to remove old kv cache entries (id: %v, keep: %v discard: %v): %w", slot.Id, numKeep, discard, err)
+			slog.Debug("kv cache removal failed, clearing cache and returning inputs for reprocessing",
```

Comment:
In most cases, this simply means the model doesn't support the operation, so we might want to soften the error message to indicate that it isn't necessarily unexpected. If you want to be sure this is the case, you can check for `kvcache.ErrNotSupported` on the Ollama engine.

The old engine doesn't exactly give this information but I believe that model not supporting the operation is the only reason why it fails. That is true for both cases of `shiftFailed` - they pretty much mean the same thing, just for different root causes.